tests/units/fastsync/commons/test_split_gzip.py

```diff
@@ -0,0 +1,149 @@
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00005')
+        # chunk seq should increase and size should reset if current_chunk_size_mb greater than chunk_size_mb
+        gzip_splitter.current_chunk_size_mb = 1050
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00006')
+        self.assertEqual(gzip_splitter.current_chunk_size_mb, 0)
+        # chunk seq should not increase further if chunk_seq equals to split_file_max_chunks
+        gzip_splitter.chunk_seq = 20
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00020')
+
+    def test_write_with_no_split(self):
+        """
+        Write gzip without splitting it and reading it
+        """
+        # max_chunk = 0 should create a file with no splitting
+        with split_gzip.SplitGzipFile(self.filename, 'wb', max_chunks=0) as f_write:
+            f_write.write(DATA_WITH_100_BYTES * 50)
+
+        with gzip.open(self.filename, 'rb') as f_read:
+            file_content = f_read.read()
+
+        self.assertEqual(file_content, DATA_WITH_100_BYTES * 50)
+
+    def test_write_with_single_chunk(self):
+        """
+        Write all data into one chunk
+        """
+        # test data fits into one chunk
+        with split_gzip.SplitGzipFile(self.filename, 'wb', chunk_size_mb=1000, max_chunks=20) as f_write:
+            f_write.write(DATA_WITH_100_BYTES * 50)
+
+        with gzip.open(f'{self.filename}.part00001', 'rb') as f_read:
+            file_content = f_read.read()
+
+        self.assertEqual(file_content, DATA_WITH_100_BYTES * 50)
+
+    def test_write_with_multiple_chunks(self):
+        """
+        Write data into multiple gzip files
+        """
+        # test data fits into one chunk
+        with split_gzip.SplitGzipFile(self.filename, 'wb',
+                                      chunk_size_mb=split_gzip.SplitGzipFile._bytes_to_megabytes(200),
+                                      max_chunks=20,
+                                      est_compr_rate=1) as f_write:
+            # Write 1100 bytes of test data
+            for _ in itertools.repeat(None, 11):
+                f_write.write(DATA_WITH_100_BYTES)
+
+        # Result should be in 6 gzip files
+        with gzip.open(f'{self.filename}.part00001', 'rb') as f_read:
+            self.assertEqual(f_read.read(), DATA_WITH_100_BYTES * 2)
+        with gzip.open(f'{self.filename}.part00002', 'rb') as f_read:
+            self.assertEqual(f_read.read(), DATA_WITH_100_BYTES * 2)
+        with gzip.open(f'{self.filename}.part00003', 'rb') as f_read:
+            self.assertEqual(f_read.read(), DATA_WITH_100_BYTES * 2)
+        with gzip.open(f'{self.filename}.part00004', 'rb') as f_read:
+            self.assertEqual(f_read.read(), DATA_WITH_100_BYTES * 2)
+        with gzip.open(f'{self.filename}.part00005', 'rb') as f_read:
+            self.assertEqual(f_read.read(), DATA_WITH_100_BYTES * 2)
+        # Last chunk should be smaller
```

Comment:
Can you add a similar but where you reach the `max_chunks` but the last file size is greater than the `chunk_size_mb`?